bin/tx-prover/src/proxy/mod.rs

```diff
@@ -176,8 +333,28 @@ impl ProxyHttp for LoadBalancer {
     where
         Self::CTX: Send + Sync,
     {
-        let client_addr = session.client_addr();
-        let user_id = client_addr.map(|addr| addr.to_string());
+        // Extract the client address early
+        let client_addr = match session.client_addr() {
+            Some(addr) => addr.to_string(),
+            None => {
+                return create_response_with_error_message(
+                    session,
+                    "No socket address".to_string(),
+                )
+                .await;
+            },
+        };
+
+        info!("Client address: {:?}", client_addr);
+
+        // Special handling for localhost
+        if client_addr.contains("127.0.0.1") {
```

Comment:
I would probably create something like `LOCALHOST_ADDR` constant.

Reply:
Replaced it with the constant.